crates/nu-plugin-protocol/src/lib.rs

```diff
@@ -549,6 +549,23 @@ pub enum EngineCall<D> {
         /// Whether to redirect stderr from external commands
         redirect_stderr: bool,
     },
+    /// Evaluate a closure with stream input/output, cloning engine state for concurrent execution
+    ///
+    /// Unlike `EvalClosure`, this variant clones the engine state and stack, allowing the closure
+    /// to be evaluated concurrently without blocking other engine calls. This is useful for
+    /// plugins that need to handle multiple requests simultaneously (e.g., HTTP servers).
+    EvalClosureCloned {
```

Comment:
bikeshedding here, the name implies how it's created versus what it's meant to do. why would the end user care that it's cloned? maybe something like `EvalClosureParallel` or `EvalClosureWithoutBlocking` or something that tells the use with the name what it's supposed to be doing.